src/main/java/org/ethereum/beacon/discovery/liveness/LivenessChecker.java

```diff
@@ -54,6 +64,7 @@ private void sendPing(final NodeRecord node) {
             (__, error) -> {
               if (error != null) {
                 LOG.trace("Liveness check failed for node {}", node, error);
+                ignoredNodes.add(node);
               }
```

Comment:
We should move this into the `synchronized` block below and then `ignoreNodes` won't need to be thread safe.